crates/matrix-sdk-sqlite/src/crypto_store.rs

```diff
@@ -151,16 +151,18 @@ impl SqliteCryptoStore {
         Ok(rmp_serde::from_slice(&decoded)?)
     }
 
-    fn deserialize_pickled_inbound_group_session(
+    fn deserialize_and_unpickle_inbound_group_session(
         &self,
-        value: &[u8],
+        value: Vec<u8>,
         backed_up: bool,
-    ) -> Result<PickledInboundGroupSession> {
-        let mut pickle: PickledInboundGroupSession = self.deserialize_value(value)?;
+    ) -> Result<InboundGroupSession> {
+        let mut pickle: PickledInboundGroupSession = self.deserialize_value(&value)?;
+
         // backed_up SQL column is source of truth, backed_up field in pickle
         // needed for other stores though
```

Comment:
It would be nice if this comment mentioned that the pickle is not updated when the key is backed up, which is *why* the column is the source of truth.

Reply:
Done in [dec768c](https://github.com/matrix-org/matrix-rust-sdk/pull/3863/commits/dec768c4fd7e44a9030226eb7fb47e3a67572e05)